libsrc/psg/wyz/wyzplay.asm

```diff
@@ -980,19 +998,26 @@ RS_CANALC:
     LD      (IX+4),B
     LD      (IX+5),C
     LD      (IX+10),D
+poneruidomixer:		;este punto es comun a los 3 canales
+    ld a,e	;cargamos la máscara para el mixer
+    and (ix+7)	;lo aplicamos desde el buffer de registros del ay
```

Comment:
@jsmolina we need to ensure that the top two bits of register 7 are 10b or MSX machines start to explode.

Reply:
I've been talking with original author (thEpOpE), and looks like


1. Mixer (IX+7) it's set by WYZ tracker
2. Initially, 'E' has a common mask (%11110111)
this mask only nullifies one bit. C channel bit.

For channel B, register is rotated twice, cyclically. 
So the bit that is kicked our on the right is added from the left.

For channel A, register is rotated once.

So, for channel A: register E = %11110111
in channel B: register E= %11101111
in channel A: register E = %11011111

this mask is applied in three instructions
```
poneruidomixer:    ;este punto es comun a los 3 canales
      ld a,e  ;cargamos la máscara para el mixer
      and (ix+7)  ;lo aplicamos desde el buffer de registros del ay
      ld (ix+7),a  ;dejamos el resultado en el buffer de registros del ay
```

The problem are as you said with 2 first bits (bit 7 and bit 6). Rotations are cyclic, top bits are never altered.
These two high bits are enabling/disabling the two pins IN/OUT that AY has, so they must be always preserved, as ZX Spectrum also uses them.

Player also considers them here:
```
FIN_NOPLAYER:
            LD      A,10111000B
            LD      [PSG_REG+7],A
            RET
```

Reply:
Great comment, thank you